### PR TITLE
 Expand ARM output into Pipeline variables

### PR DESCRIPTION
**Pull request by TomBonnerAtTFL**
Hi there Microsoft! I've added code that expands the returned Outputs object into Pipeline variables so they can be used easily in scripts. This fixes issue #13032.

**Task name**: AzureResourceGroupDeploymentV2

**Description**: Added code to expand the Outputs into more accessible variables

**Documentation changes required:** Yes

**Added unit tests:** No

**Attached related issue:** Original issue can be found [here](https://github.com/microsoft/azure-pipelines-tasks/issues/13032).

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

**Comment by TomBonnerAtTFL** [comment]
Hi @bishal-pdMSFT, is there anything else you'd like me to do to progress this PR?

**Comment by bishal-pdMSFT** [comment]
@TomBonnerAtTFL the changes you have done is to v2 version of task. while that is good, please make same changes in v3 task as well - `AzureResourceManagerTemplateDeploymentV3`

File: Tasks/AzureResourceGroupDeploymentV2/task.json
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/task.json:17]: Please ensure that this version is still greater than latest as the PR is a bit old now.

**Comment by bishal-pdMSFT** [comment]
Overall the changes look good to me. Please take a look at couple of comments.

File: Tasks/AzureResourceGroupDeploymentV2/task.json
Comment (TomBonnerAtTFL) [reply Tasks/AzureResourceGroupDeploymentV2/task.json]: I have set both to the current sprint number of 179.

**Comment by TomBonnerAtTFL** [comment]
Hi @bishal-pdMSFT, I have made your recommended changes. My apologies for not seeing the `AzureResourceManagerTemplateDeploymentV3` task, I have added my changes to that as well now.

**Review by bishal-pdMSFT — APPROVED** [review]

File: Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts:567]: can you please add a console log to print the name of the variable being set ? Similar to L575.

**Comment by bishal-pdMSFT** [comment]
@TomBonnerAtTFL Looks good to me. Left one comment about logging. After that, I can go ahead and merge the PR.

**Comment by TomBonnerAtTFL** [comment]
@bishal-pdMSFT I've added the console lines to both task's output. Good spot.

**Comment by TomBonnerAtTFL** [comment]
Hi @bishal-pdMSFT I have hopefully solved all your comments and it is good to merge if you see fit.

**Review by bishal-pdMSFT — APPROVED** [review]

**Comment by bishal-pdMSFT** [comment]
@TomBonnerAtTFL merged 😀. Thank you so much for your contribution and making us better 🎉

**Comment by TomBonnerAtTFL** [comment]
I was worried my change had broke your build! But it seems that there is a bug in your CI.

In step "Remove CodeSignSummary file from package" you are seeing the following error:

image

This means the publish eventually fails due to the extra file

**Pull request merged**
